frontend/src/components/core/ReportView/ReportView.tsx

```diff
@@ -73,8 +77,20 @@ function ReportView(props: ReportViewProps): ReactElement {
     widgetsDisabled,
     uploadClient,
     componentRegistry,
+    s4aCommunication,
   } = props
 
+  React.useEffect(() => {
+    const listener = (): void => {
+      s4aCommunication.sendMessage({
```

Comment:
I think if we add the hoc here, we are going to initiate the 'ready' procedure twice, I would just import sendS4AMessage as we need it only for that.